### PR TITLE
Adding label support

### PR DESCRIPTION
**Pull request by bathina2**
## Change Overview

This add support to label volume snapshots as we take them.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Issues

- #XXX

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/kube/snapshot/snapshot.go [hunk 1]
```diff
@@ -53,7 +53,7 @@ type Snapshotter interface {
 	// 'pvcName' is the name of the PVC of which we will take snapshot. It must be in the same namespace 'ns'.
 	// 'waitForReady' will block the caller until the snapshot status is 'ReadyToUse'.
 	// or 'ctx.Done()' is signalled. Otherwise it will return immediately after the snapshot is cut.
-	Create(ctx context.Context, name, namespace, pvcName string, snapshotClass *string, waitForReady bool) error
+	Create(ctx context.Context, name, namespace, pvcName string, snapshotClass *string, waitForReady bool, labels map[string]string) error
```
Comment (onkarbhat) [inline pkg/kube/snapshot/snapshot.go:56]: Update the comment with info about the new labels argument ?

File: pkg/kube/snapshot/snapshot.go [hunk 2]
```diff
@@ -102,6 +102,8 @@ type Snapshotter interface {
 	// WaitOnReadyToUse will block until the Volumesnapshot in namespace 'namespace' with name 'snapshotName'
 	// has status 'ReadyToUse' or 'ctx.Done()' is signalled.
 	WaitOnReadyToUse(ctx context.Context, snapshotName, namespace string) error
+	// List will list the volumesnapshots in a namespace that match search
```
Comment (onkarbhat) [inline pkg/kube/snapshot/snapshot.go:105]: Should the comment say  "List will.....that match with the given labels"  . What is the behavior when no labels are provided ? Will it return no snapshots or all snapshots ?

File: pkg/kube/snapshot/snapshot_test.go [hunk 3]
```diff
@@ -403,7 +403,10 @@ func (s *SnapshotTestSuite) testVolumeSnapshot(c *C, snapshotter snapshot.Snapsh
 
 	snapshotName := snapshotNamePrefix + strconv.Itoa(int(time.Now().UnixNano()))
 	wait := true
-	err = snapshotter.Create(ctx, snapshotName, s.sourceNamespace, pvc.Name, snapshotClass, wait)
+	label := map[string]string{
+		"snapshottest": "testlabel",
+	}
+	err = snapshotter.Create(ctx, snapshotName, s.sourceNamespace, pvc.Name, snapshotClass, wait, label)
```
Comment (onkarbhat) [inline pkg/kube/snapshot/snapshot_test.go:409]: Please add a test where Create is called with a label map that is not nil and with no keys in it. That will help clarify the behavior of List when such a map is provided ( I am guessing it will return nothing ? ) .

**Review by onkarbhat — APPROVED** [review]
LGTM. Please add the test case suggested in snapshot_test.go to clarify the behavior of List

File: pkg/kube/snapshot/snapshot.go (hunk 2 above)
Comment (bathina2) [reply pkg/kube/snapshot/snapshot.go:105]: All snapshots in the namespace.

File: pkg/kube/snapshot/snapshot_test.go (hunk 3 above)
Comment (bathina2) [reply pkg/kube/snapshot/snapshot_test.go:409]: No it will create it without any labels. List creates a matcher out of the given labels.
